### PR TITLE
Added Filter to index supports

### PR DESCRIPTION
**Pull request by guilleLara**
This endpoint adds pagination and the posibility to apply a searching filter (both email and first_name) to the existing index of supports.

Trello card : https://trello.com/c/fY5pjwcA/161-agregar-busqueda-a-la-tabla-de-soportes

Also, I did some style and spell check on the README file.

For more information of the endpoint please see README.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -258,77 +258,130 @@ This endpoint can be used by an administrator or support.
...
+This endpoint returns a paginated list of available support members for a site. This list contains both, confirmed and not yet confirmed supports.
+ - Notes : 
+  1. Not confirmed supports will come with null first_name and last_name. Also the confirmed_at param will be null.
+  2. Not confirmed supports will appear first in the result list.
+  
 Only administrator users are allowed to use this functionality.
 This endpoint is for Dashboard use only.
 The administrator user must be logged-in into the Dashboard. 
 
+Mandatory body fields:
```
Comment (petterboussard) [inline README.md]: This param is optional, right?

File: README.md [hunk 2]
```diff
@@ -258,77 +258,130 @@ This endpoint can be used by an administrator or support.
...
+- Search -> String to apply filtering to the supports list, the search is done by email or first_name.
+- Items -> If provided, then the pagination will be done with that number of items per page, this number defaults to 20 items per page if not provided.
+
 **Headers**:
 Content-type: application/json  
 AUTH HEADERS NEEDED! You can get this  headers by calling the endpoint {app_domain}/login  
 
-**Body**
-No body is needed for this endpoint  
+**Request Body**
```
Comment (petterboussard) [inline README.md]: This param goes in the query params, not in the body. This should be changed.

File: README.md [hunk 3]
```diff
@@ -258,77 +258,130 @@ This endpoint can be used by an administrator or support.
...
+**Request Body**  
+{
+  "page": number
+}
 
-**Response Schema Example**
+**Request Body (Optional Fields)**  
+{
+  "search": string,
+  "items": number
```
Comment (petterboussard) [inline README.md]: This param also goes in the query params.

File: README.md [hunk 4]
```diff
@@ -258,77 +258,130 @@ This endpoint can be used by an administrator or support.
...
 ``` 
 
 **Description**
-This endpoint returns a list of deleted support members for a site.
+This endpoint returns a paginated list of deleted support members for a site.
 Only administrator users are allowed to use this functionality.
 This endpoint is for Dashboard use only.
 The administrator user must be logged-in into the Dashboard. 
 
+Mandatory body fields:
```
Comment (petterboussard) [inline README.md]: Same changes as the above comments.

File: app/controllers/api/v1/sites_controller.rb [hunk 5]
```diff
@@ -67,49 +66,26 @@ def update_site
...
-              email: s.email.split('|', 2).second,
-              first_name: s.first_name,
-              last_name: s.last_name,
-              deleted_at: s.deleted_at
-            }
-          end
-          render json: supports_list, status: :ok
+          supports = site.users.supports.with_deleted
+          supports = supports.where('email ILIKE ? or first_name ILIKE ?', "%#{search}%", "%#{search}%") if search.present?
+          pagy, @supports_list = pagy(supports, items: params[:items])
```
Comment (petterboussard) [inline app/controllers/api/v1/sites_controller.rb]: What happens if supports is empty?

File: app/controllers/api/v1/sites_controller.rb [hunk 6]
```diff
@@ -67,49 +66,26 @@ def update_site
...
-            {
-              id: s.id,
-              email: s.email.split('|', 2).second,
-              first_name: s.first_name,
-              last_name: s.last_name,
-              deleted_at: s.deleted_at
-            }
-          end
-          render json: supports_list, status: :ok
+          supports = site.users.supports.with_deleted
```
Comment (petterboussard) [inline app/controllers/api/v1/sites_controller.rb]: The documentation of this endpoint does not state that deleted supports are going to be returned.
Can someone please update the README.md file.

File: spec/requests/sites_controller_spec.rb [hunk 7]
```diff
@@ -13,71 +13,69 @@
   let!(:admin2) { create(:user, site: site2, role: 'admin', password: 'password') }
   let!(:support) { create(:user, site: site1, role: 'support', password: 'password') }
   let!(:deleted_supp) { create(:user, site: site1, role: 'support', password: 'password') }
+  let!(:search_support) { create(:user, site: site1, role: 'support', first_name: 'FIRST NAME') }
```
Comment (petterboussard) [inline spec/requests/sites_controller_spec.rb]: first_name FIELD should be populated using Faker gem.

File: spec/requests/sites_controller_spec.rb [hunk 8]
```diff
@@ -13,71 +13,69 @@
...
+  let!(:search_deleted_supp) { create(:user, site: site1, role: 'support', first_name: 'DELETED USER') }
```
Comment (petterboussard) [inline spec/requests/sites_controller_spec.rb]: Same here with the first_name.

File: spec/requests/sites_controller_spec.rb [hunk 9]
```diff
@@ -13,71 +13,69 @@
...
   after(:each) { Chat.destroy_all }
   after(:each) { User.with_deleted.each(&:really_destroy!) }
   after(:each) { Site.destroy_all }
+
   let(:request_list_supports) do
     login(admin1)
     auth_params = get_auth_params_from_login_response_headers(response)
     site1.users << support
-    get api_v1_site_supports_path, headers: auth_params
+    get api_v1_site_supports_path, params: { page: 1 }, headers: auth_params
```
Comment (petterboussard) [inline spec/requests/sites_controller_spec.rb]: If the param isn't provided the first page will be returned, right? 
Why is this necessary?

File: spec/requests/sites_controller_spec.rb [hunk 10]
```diff
@@ -13,71 +13,69 @@
...
-                                              'confirmation_sent_at', 'invitation_expired', 'confirmed_at')
+                                              'confirmation_sent_at', 'confirmed_at', 'invitation_expired', 'deleted_at')
         end
       end
 
       it 'should return a list with only the users whose role is support' do
         request_list_supports
-        hash = JSON.parse(response.body)
+        hash = JSON.parse(response.body).values.first
         hash.each do |obj|
```
Comment (petterboussard) [inline spec/requests/sites_controller_spec.rb]: Lets change this **obj** variable and use a meaningful name.

File: spec/requests/sites_controller_spec.rb [hunk 11]
```diff
@@ -104,53 +102,34 @@
...
-    context 'requesting the supports list of a site but the user is not an administrator' do
-      it 'response should have 403 status forbidden' do
-        request_deleted_supports_forbidden
-        expect(response).to have_http_status(403)
-        expect(JSON.parse(response.body).keys).to contain_exactly('errors')
-        expect(JSON.parse(response.body).values).to eq([['Only administrator users are allowed']])
+      it 'should have proper values for returned element(s)' do
+        request_search_supports
+        hash = JSON.parse(response.body).values.first
+        hash.each do |obj|
```
Comment (petterboussard) [inline spec/requests/sites_controller_spec.rb]: **OBJ** again

File: README.md (hunk 1 above)
Comment (jicc96) [reply README.md]: It is mandatory if not present it returns a 400 error

File: README.md (hunk 2 above)
Comment (jicc96) [reply README.md]: done

File: README.md (hunk 3 above)
Comment (jicc96) [reply README.md]: done

File: README.md (hunk 4 above)
Comment (jicc96) [reply README.md]: Same as above

File: app/controllers/api/v1/sites_controller.rb (hunk 5 above)
Comment (jicc96) [reply app/controllers/api/v1/sites_controller.rb]: This is not the case, if it's empty returns an empty record

File: spec/requests/sites_controller_spec.rb (hunk 7 above)
Comment (jicc96) [reply spec/requests/sites_controller_spec.rb]: done

File: spec/requests/sites_controller_spec.rb (hunk 8 above)
Comment (jicc96) [reply spec/requests/sites_controller_spec.rb]: done

File: spec/requests/sites_controller_spec.rb (hunk 9 above)
Comment (jicc96) [reply spec/requests/sites_controller_spec.rb]: I don't know i did not write this test but what you say is true

File: app/controllers/api/v1/sites_controller.rb (hunk 6 above)
Comment (jicc96) [reply app/controllers/api/v1/sites_controller.rb]: done

File: spec/requests/sites_controller_spec.rb (hunk 9 above)
Comment (jicc96) [reply spec/requests/sites_controller_spec.rb]: Remember that it is a mandatory field

File: spec/requests/sites_controller_spec.rb (hunk 10 above)
Comment (jicc96) [reply spec/requests/sites_controller_spec.rb]: done

File: spec/requests/sites_controller_spec.rb (hunk 11 above)
Comment (jicc96) [reply spec/requests/sites_controller_spec.rb]: done
